pulsar-functions/worker/src/main/java/org/apache/pulsar/functions/worker/rest/api/FunctionsImpl.java

```diff
@@ -699,6 +707,9 @@ private void validateTriggerRequestParams(String tenant,
         if (functionName == null) {
             throw new IllegalArgumentException("Function Name is not provided");
         }
+        if (topic == null) {
```

Comment:
topic has to be optional. It is  only required for functions that have more than one input topic

Reply:
sure we can add that, but since there are already a big number of changes, lets do that in a subsequent PR.  Lets first just get the basics in place